django/db/migrations/autodetector.py

```diff
@@ -1305,14 +1305,19 @@ def suggest_name(cls, ops):
         into the fallback name to avoid VCS conflicts if possible.
         """
         if len(ops) == 1:
-            if isinstance(ops[0], operations.CreateModel):
-                return ops[0].name_lower
-            elif isinstance(ops[0], operations.DeleteModel):
-                return "delete_%s" % ops[0].name_lower
-            elif isinstance(ops[0], operations.AddField):
-                return "%s_%s" % (ops[0].model_name_lower, ops[0].name_lower)
-            elif isinstance(ops[0], operations.RemoveField):
-                return "remove_%s_%s" % (ops[0].model_name_lower, ops[0].name_lower)
+            op = ops[0]
+            if isinstance(op, operations.CreateModel):
+                return op.name_lower
+            elif isinstance(op, operations.DeleteModel):
+                return "delete_%s" % op.name_lower
+            elif isinstance(op, operations.AddField):
+                return "%s_%s" % (op.model_name_lower, op.name_lower)
+            elif isinstance(op, operations.RemoveField):
+                return "remove_%s_%s" % (op.model_name_lower, op.name_lower)
+            elif isinstance(op, operations.AddConstraint):
+                return "add_%s_%s" % (op.model_name_lower, op.constraint.name.lower())
+            elif isinstance(op, operations.RemoveConstraint):
+                return "remove_%s_%s" % (op.model_name_lower, op.name.lower())
```

Comment:
It'd be great if this was implemented as an `Operation` method that defaults to returning `None` and overridden when available. It would make testing easier and remove this chain of instance checks.

Reply:
You're right.